### PR TITLE
Feat/security enterprise

### PR DESCRIPTION
**Pull request by dominikpalatynski**
## Summary

This PR introduces the enterprise security module, adding a centralized security layer for authentication hardening and admin control. It brings multi-factor authentication support for users, MFA enforcement policies for administrators, sudo-style re-authentication for sensitive actions, and user security management flows, so organizations can enforce stronger account protection without custom security logic across the app.

### CODE REVIEW

File: apps/mercato/src/app/(backend)/backend/layout.tsx [hunk 1]
```diff
@@ -389,36 +388,34 @@ export default async function BackendLayout({ children, params }: { children: Re
     <>
       <Script async src="https://w.appzi.io/w.js?token=TtIV6" strategy="afterInteractive" />
       <I18nProvider locale={locale} dict={dict}>
-        <ComponentOverridesBootstrap>
```
Comment (pkarw) [inline apps/mercato/src/app/(backend)/backend/layout.tsx:392]: Why has this been removed? Is it 100% intentional? I suppost it might disable the components override mechanism from the UMES spec

File: apps/mercato/src/components/AppProviders.tsx [hunk 2]
```diff
@@ -7,6 +7,7 @@ import { I18nProvider } from '@open-mercato/shared/lib/i18n/context'
 import { ThemeProvider, FrontendLayout, QueryProvider, AuthFooter } from '@open-mercato/ui'
 import { ClientBootstrapProvider } from '@/components/ClientBootstrap'
 import { GlobalNoticeBars } from '@/components/GlobalNoticeBars'
+import { ComponentOverridesBootstrap } from '@/components/ComponentOverridesBootstrap'
```
Comment (pkarw) [inline apps/mercato/src/components/AppProviders.tsx:10]: OK, it's been moved here, but what was the reason?

File: apps/mercato/src/bootstrap.ts [hunk 3]
```diff
@@ -53,11 +53,19 @@ import { messageTypes } from '@/.mercato/generated/message-types.generated'
 import { messageObjectTypes } from '@/.mercato/generated/message-objects.generated'
 import { registerMessageTypes } from '@open-mercato/core/modules/messages/lib/message-types-registry'
 import { registerMessageObjectTypes } from '@open-mercato/core/modules/messages/lib/message-objects-registry'
+import { securityMfaProviderEntries } from '@/.mercato/generated/security-mfa-providers.generated'
```
Comment (pkarw) [inline apps/mercato/src/bootstrap.ts]: these imports can't stay here as they're breaking the app when no`enterprise` module is enabled (99,9% cases  - I mean only some enterprise users are buying and enabling these features) - we must find a way to inject the logic from the enterprise module here so `boostrap` does not know anything about enterprise modules please @dominikpalatynski

Comment (pkarw) [reply apps/mercato/src/bootstrap.ts]: This line causes the #966 - please fix it

File: apps/mercato/.env.example [hunk 4]
```diff
@@ -37,6 +37,27 @@ OM_ENABLE_ENTERPRISE_MODULES=false
...
+# Enterprise security module (MFA, passkeys, sudo)
+# Active when enterprise modules are enabled and the security module is installed.
+# Optional dedicated signing secret for MFA setup tokens.
+# Falls back to AUTH_JWT_SECRET, AUTH_SECRET, or JWT_SECRET when left blank.
+# SECURITY_MFA_SETUP_SECRET=change-me-mfa-setup-secret
+# Enable enterprise security module (default: false)
+# Requires OM_ENABLE_ENTERPRISE_MODULES=true
+OM_ENABLE_ENTERPRISE_MODULES_SECURITY=false
+
+SECURITY_TOTP_ISSUER="Open Mercato"
```
Comment (pkarw) [inline apps/mercato/.env.example]: should start with the `OM_`

File: packages/cli/src/lib/generators/module-registry.ts [hunk 5]
```diff
@@ -407,6 +407,10 @@ export async function generateModuleRegistry(options: ModuleRegistryOptions): Pr
   const analyticsChecksumFile = path.join(outputDir, 'analytics.generated.checksum')
   const transFieldsOutFile = path.join(outputDir, 'translations-fields.generated.ts')
   const transFieldsChecksumFile = path.join(outputDir, 'translations-fields.generated.checksum')
+  const securityMfaProvidersOutFile = path.join(outputDir, 'security-mfa-providers.generated.ts')
```
Comment (pkarw) [inline packages/cli/src/lib/generators/module-registry.ts]: This centralized module can not know anything about the `enterprise` module - the enterprise module should be able to be just disabled, and now it's not bc. this depncesis are centralized

If needed, please refactor the `module-registry` to be able to register the `generators` from the module level. This would be a nice extensibility point, wdyt @dominikpalatynski @pat-lewczuk ?

File: packages/create-app/template/src/bootstrap.ts [hunk 6]
```diff
@@ -53,11 +53,19 @@ import { messageTypes } from '@/.mercato/generated/message-types.generated'
 import { messageObjectTypes } from '@/.mercato/generated/message-objects.generated'
 import { registerMessageTypes } from '@open-mercato/core/modules/messages/lib/message-types-registry'
 import { registerMessageObjectTypes } from '@open-mercato/core/modules/messages/lib/message-objects-registry'
+import { securityMfaProviderEntries } from '@/.mercato/generated/security-mfa-providers.generated'
```
Comment (pkarw) [inline packages/create-app/template/src/bootstrap.ts]: Identically same issues like for the `apps/mercato`

**Review by pkarw — CHANGES_REQUESTED** [review]
# Code Review: feat/security-enterprise

## Summary

This branch introduces security enhancements (MFA context, sudo services, component sanitization, readiness probes), catalog UOM/price validation, auth profile improvements, dev container setup, release scripts, and database migration safety improvements. Overall a solid set of changes, with one test failure that must be fixed before merge.

## CI/CD Verification

| Gate | Status | Notes |
|------|--------|-------|
| `yarn build:packages` | ✅ PASS | 16/16 packages built |
| `yarn generate` | ✅ PASS | All generators completed |
| `yarn build:packages` (rebuild) | ✅ PASS | Verified via initial build |
| `yarn i18n:check-sync` | ✅ PASS | All 4 locales in sync |
| `yarn i18n:check-usage` | ⚠️ WARN | 35 missing keys, 3638 unused (non-blocking in CI) |
| `yarn typecheck` | ✅ PASS | 16/16 packages pass |
| `yarn test` | ❌ **FAIL** | 1 test timeout in `@open-mercato/cli` |
| `yarn build:app` | ✅ PASS | Next.js build successful |

## Findings

### 🔴 Critical

_None_

### 🟠 High

**H1: Unit test timeout — `integration.test.ts:124`**
- **File**: `packages/cli/src/lib/testing/__tests__/integration.test.ts:124`
- **Issue**: Test `reuses an existing environment when /login returns a redirect status other than 302` times out at the default 5000ms. The test calls `tryReuseExistingEnvironment()` which invokes `hasBuildInputChangesSince()`, scanning the entire project tree for file mtime changes. Adjacent tests at lines 80 and 163 have explicit `20000` timeouts but this test does not.
- **Fix**: Add a timeout parameter: `}, 20000)` at line 161 (matching the pattern of surrounding tests).

### 🟡 Medium

**M1: Empty catch blocks in product page**
- **File**: `packages/core/src/modules/catalog/backend/catalog/products/[id]/page.tsx`
- **Issue**: Several `catch` blocks in `loadVariants` and variant price loading return empty arrays silently without logging. While defensive, this can mask real errors during development.
- **Suggestion**: Add at minimum `console.error` or pass error to a logging utility.

**M2: i18n missing keys (35 keys)**
- While non-blocking in CI, 35 missing i18n keys should be addressed. Some may be from this branch's new features.

### 🟢 Low

**L1: Test file TC-AUTH-020 — consider adding MFA-enabled scenario**
- `TC-AUTH-020.spec.ts` only tests login without MFA interceptors. Consider a complementary test for when MFA is configured.

**L2: priceValidation.ts could be shared**
- `packages/core/src/modules/catalog/lib/priceValidation.ts` is well-isolated with zero dependencies. If other modules (sales, invoices) need price validation, consider extracting to `@open-mercato/shared`. Not urgent — current placement is architecturally correct for catalog-only use.

## Backward Compatibility

- [x] No contract surface removed or renamed without deprecation bridge
- [x] No event IDs renamed or removed
- [x] No widget injection spot IDs renamed or removed
- [x] No API route URLs renamed or removed
- [x] No existing response schema fields removed (profile API adds `roles` — additive)
- [x] No database columns/tables renamed or removed
- [x] No DI service names renamed or removed
- [x] No ACL feature IDs renamed or removed
- [x] No public import paths removed without re-export bridge
- [x] No required type fields removed or narrowed
- [x] No function signatures changed in a breaking way
- [x] Deprecation protocol followed (if applicable)

## Checklist

- [x] No `any` types introduced
- [x] All API routes export `openApi`
- [x] Validators in `data/validators.ts` (not inline)
- [x] Tenant isolation: queries filter by `organization_id`
- [x] No hardcoded user-facing strings
- [x] CRUD routes use `makeCrudRoute` with `indexer`
- [x] Events declared in `events.ts` before emitting
- [x] Workers/subscribers export `metadata`
- [x] Custom fields use `collectCustomFieldValues()`
- [x] `modules:prepare` needed after file additions — verified via `yarn generate`
- [x] No cross-module ORM relationships
- [x] Encryption helpers used instead of raw `em.find`
- [x] Forms use `CrudForm`, tables use `DataTable`
- [x] `apiCall` used instead of raw `fetch`
- [x] ACL features mirrored in `setup.ts` `defaultRoleFeatures`
- [x] Behavior changes covered by unit and/or integration tests
- [ ] No empty `catch` blocks — **M1**: some empty catches in product page
- [x] New migrations are scoped to intended entities only
- [x] No conflicting spec IDs

## Change Summary by Area

| Area | Changes | Assessment |
|------|---------|------------|
| **Auth security** | Profile API adds `roles`, password change with current-password verification, integration tests TC-AUTH-019/020 | ✅ Solid security practices |
| **Catalog** | UOM support, price validation lib, variant price forms, TC-CAT-015 integration test | ✅ Well-structured, comprehensive edge case handling |
| **CrudForm** | New field types (datepicker/datetime/time), replacement handles, readOnly, submitIcon | ✅ Fully backward compatible — all additive |
| **CLI/DB** | Migration safety (SQL injection prevention, `current_schema()`), version alignment script | ✅ Important hardening |
| **Release scripts** | Template sync check, snapshot releases, version alignment | ✅ Good release hygiene |
| **Dev container** | Full devcontainer setup (Dockerfile, docker-compose, lifecycle scripts) | ✅ New infrastructure, isolated |
| **Packages** | gateway-stripe and sync-akeneo scaffolding | ✅ Follows conventions |

## Blocking Issue

> **Fix H1 (test timeout) before merge** — CI will fail on `yarn test`. The fix is a one-line change adding `, 20000` timeout to the test at `packages/cli/src/lib/testing/__tests__/integration.test.ts:161`.

File: apps/mercato/src/components/AppProviders.tsx (hunk 2 above)
Comment (dominikpalatynski) [reply apps/mercato/src/components/AppProviders.tsx:10]: Yes, exactly. The primary driver was that MFA challenge/enrollment pages live in the (frontend) route group. Since backend/layout.tsx only covers /backend/*, those pages had no override support before and we can't inject MFA verification on login page

**Review by pkarw — APPROVED** [review]

**Review by pkarw — APPROVED** [review]
